ui/app/AppLayouts/Chat/views/CommunitySettingsView.qml

```diff
@@ -250,7 +252,57 @@ StatusSectionLayout {
-                assetsModel: rootStore.assetsModel
+
+                // temporary solution to provide icons for assets, similar
+                // method is used in wallet (constructing filename from asset's
+                // symbol) and is intended to be replaced by more robust
+                // solution soon.
+                assetsModel: SortFilterProxyModel {
+                    sourceModel: rootStore.assetsModel
+
+                    proxyRoles: ExpressionRole {
+
+                        // list of symbols for which pngs are stored to avoid
+                        // accessing not existing resources and providing
+                        // default icon
+                        readonly property var pngs: [
+                            "aKNC", "AST", "BLT", "CND", "DNT", "EQUAD", "HEZ", "LOOM", "MTH",
+                            "PAY", "RCN", "SALT", "STRK", "TRST", "WBTC", "AKRO", "aSUSD", "BLZ",
+                            "COB", "DPY", "ETH2x-FLI", "HST", "LPT", "MTL", "PBTC", "RDN", "SAN",
+                            "STT", "TRX", "WETH", "0-native", "aLEND", "ATMChain", "BNB", "COMP",
+                            "DRT", "ETHOS", "HT", "LRC", "MYB", "PLR", "renBCH", "SNGLS", "STX",
+                            "TUSD", "WINGS", "0XBTC", "aLINK", "aTUSD", "BNT", "CUSTOM-TOKEN",
+                            "DTA", "ETH", "ICN", "MANA", "NEXO", "POE", "renBTC", "SNM", "SUB",
+                            "UBT", "WTC", "1ST", "aMANA", "aUSDC", "BQX", "CVC", "EDG", "EVX",
+                            "ICOS", "MCO", "NEXXO", "POLY", "REN", "SNT", "SUPR", "UKG", "XAUR",
+                            "aBAT", "AMB", "aUSDT", "BRLN", "DAI", "EDO", "FUEL", "IOST", "MDA",
+                            "NMR", "POWR", "renZEC", "SNX", "SUSD", "UNI", "XPA", "ABT", "aMKR",
+                            "aWBTC", "BTM", "DATA", "EKG", "FUN", "KDO", "MET", "NPXS", "PPP",
+                            "REP", "SOCKS", "TAAS", "UPP", "XRL", "aBUSD", "AMPL", "aYFI", "BTU",
+                            "DAT", "EKO", "FXC", "KIN", "MFG", "OGN", "PPT", "REQ", "SPANK",
+                            "TAUD", "USDC", "XUC", "ABYSS", "ANT", "aZRX", "CDAI", "DCN", "ELF",
+                            "GDC", "KNC", "MGO", "OMG", "PT", "RHOC", "SPIKE", "TCAD", "USDS",
+                            "ZRX", "aDAI", "APPC", "BAL", "CDT", "DEFAULT-TOKEN", "EMONA", "GEN",
+                            "Kudos", "MKR", "OST", "QKC", "RLC", "SPN", "TGBP", "USDT", "ZSC",
+                            "aENJ", "aREN", "BAM", "Centra", "DGD", "ENG", "GNO", "LEND", "MLN",
+                            "OTN", "QRL", "ROL", "STORJ", "TKN", "VERI", "AE", "aREP", "BAND",
+                            "CFI", "DGX", "ENJ", "GNT", "LINK", "MOC", "PAXG", "QSP", "R",
+                            "STORM", "TKX", "VIB", "aETH", "aSNX", "BAT", "CK", "DLT", "EOS",
+                            "GRID", "LISK", "MOD", "PAX", "RAE", "SAI", "ST", "TNT", "WABI"
+                        ]
```

Comment:
How do you feel about having `Style.png()` just try to do its thing, given any symbol, and if the image can't be loaded, because it doesn't exist, use a default image instead?

That way we don't need to maintain this list of symbols here.

Reply:
we could do that but the cost is a lot of warning in the console when there is no png, and lack of replacement token image when image not found. According to Anthony the access based on `Style.png("tokens/" + symbol)` will be replaced soon anyway.

Reply:
Take the `pngs` prop out of the expression 😊

Reply:
The function too

Reply:
Ah right, nvm

Reply:
@0x-r4bbit I ment only cases like that on the screenshot:
![Screenshot from 2023-03-09 12-49-31](https://user-images.githubusercontent.com/20650004/224014959-1dda37f0-b933-476a-9848-69c76400e662.png)

If you are ok with that, I can remove that list, but personally I think that it doesn't hurt now and makes the ui fully "correct" :)

Reply:
Not sure what you're referring to?
So if a symbol icon does not exist, of course we'd still fallback to some default, which should fix the empty spots in the dropdown there (if that's what you meant)

Reply:
> Not sure what you're referring to? So if a symbol icon does not exist, of course we'd still fallback to some default, which should fix the empty spots in the dropdown there (if that's what you meant)

The problem with plain `Style.png("tokens/" + symbol)` is that providing fallback image is tricky. We would have to check `Image.status` wherever we use the icon (e.g. in "who holds" selector, in dropdown), and if the status is `Image.Error` than set the fallback image (bc, we cannot easily check up-front if the image exist in qml). So providing fallback would have to be done separately for every usage of that icon, resulting  with changes in may files, much more problematic to revert when the new solution is ready.

Reply:
Okay, so, just for the sake of the argument, I think we could change `Style.png(name)` to `Style.png(name, fallback = "")`
Then check for `fallback != ""` to apply a default token. Since `fallback` has a default value as well, we wouldn't need to touch all places. And `Style.png()` would just generally do something like: try to load  `{name}.png` and if it errors, try the same with `{fallback}.png`.

Today `Style.png()` already throws a warning when it can't load a file, so we would just add an optional fallback.

Again, just for the sake of the argument. It's your call after all.

Reply:
But `Style.png` doesn't load anything, it does only simple string concatenation. As I wrote, there is no simple way for doing that check inside Style.png(), cpp helper would be needed to actually check the resource.

Reply:
The warning is not thrown by `Style.png()` but by the Image after assigning invalid source url created by `Style.png()`

Reply:
Ahh right! Then nevermind. I confused this